verilog/tools/kythe/indexing_facts_tree_extractor.cc

```diff
@@ -1032,5 +1054,62 @@ void IndexingFactsTreeExtractor::ExtractInclude(
       content, file_path, file_list_facts_tree_, extracted_files_));
 }
 
+void IndexingFactsTreeExtractor::ExtractEnumName(
+    const verible::SyntaxTreeNode& enum_name) {
+  IndexingNodeData enum_name_data(IndexingFactType::kConstant);
+  IndexingFactNode enum_node(enum_name_data);
+
+  const SyntaxTreeLeaf& name = GetSymbolIdentifierFromEnumName(enum_name);
+  enum_node.Value().AppendAnchor(Anchor{name.get(), context_.base});
+
+  // Iterate over the children and traverse them to extract facts form inner
+  // nodes and ignore the leaves.
+  // e.g enum {RED[x] = 1, OLD=y} => explores "[x]", "=y".
+  {
+    const IndexingFactsTreeContext::AutoPop p(&facts_tree_context_, &enum_node);
+    for (const auto& child : enum_name.children()) {
+      if (child == nullptr || child->Kind() == verible::SymbolKind::kLeaf) {
+        continue;
+      }
+      Visit(verible::SymbolCastToNode(*child));
+    }
+  }
+
+  facts_tree_context_.top().NewChild(enum_node);
+}
+
+void IndexingFactsTreeExtractor::ExtractEnumTypeDeclaration(
+    const verible::SyntaxTreeNode& enum_type_declaration) {
+  // Extract enum type name.
+  const SyntaxTreeLeaf* enum_type_name =
+      GetIdentifierFromTypeDeclaration(enum_type_declaration);
+  facts_tree_context_.top().NewChild(
+      IndexingNodeData({Anchor(enum_type_name->get(), context_.base)},
+                       IndexingFactType::kVariableDefinition));
+
+  // Explore the children of this enum type to extract.
+  for (const auto& child : enum_type_declaration.children()) {
+    if (child == nullptr || child->Kind() == verible::SymbolKind::kLeaf) {
+      continue;
+    }
+    Visit(verible::SymbolCastToNode(*child));
+  }
+}
+
+void IndexingFactsTreeExtractor::ExtractTypeDeclaration(
+    const verible::SyntaxTreeNode& type_declaration) {
+  const std::vector<TreeSearchMatch> enum_types =
+      FindAllEnumTypes(type_declaration);
+  if (!enum_types.empty()) {
+    ExtractEnumTypeDeclaration(type_declaration);
+    return;
+  }
+  const std::vector<TreeSearchMatch> struct_types =
+      FindAllStructTypes(type_declaration);
+  if (!struct_types.empty()) {
+    return;
+  }
```

Comment:
This looks like an unconditional `return`.  Missing code?

Reply:
sorry missed a TODO here.